File: smash-ranks-android/app/src/main/java/com/garpr/android/models/LitePlayer.java
package com.garpr.android.models;

import android.os.Parcel;
import android.os.Parcelable;
import android.support.annotation.NonNull;

public class LitePlayer extends AbsPlayer implements Parcelable {

    public LitePlayer() {

    }

    public LitePlayer(@NonNull final String id, @NonNull final String name) {
        mId = id;
        mName = name;
    }

    @Override
    public Kind getKind() {
        return Kind.LITE;
    }

    public static final Creator<LitePlayer> CREATOR = new Creator<LitePlayer>() {
        @Override
        public LitePlayer createFromParcel(final Parcel source) {
            final LitePlayer lp = new LitePlayer();
            lp.readFromParcel(source);
            return lp;
        }

        @Override
        public LitePlayer[] newArray(final int size) {
            return new LitePlayer[size];
        }
    };

}

File: smash-ranks-android/app/src/main/java/com/garpr/android/models/LitePlayer.kt
package com.garpr.android.models

import android.os.Parcelable
import com.garpr.android.extensions.createParcel
import com.garpr.android.extensions.requireString

class LitePlayer(
        id: String,
        name: String
) : AbsPlayer(
        id,
        name
), Parcelable {

    companion object {
        @JvmField
        val CREATOR = createParcel { LitePlayer(it.requireString(), it.requireString()) }
    }

    override val kind: Kind
        get() = Kind.LITE

}
